From f05f7499b03f5b695c8d5135bb5ef11ad5b51f65 Mon Sep 17 00:00:00 2001
From: Kent Russell <kent.russell@amd.com>
Date: Tue, 21 Nov 2017 08:45:37 -0500
Subject: [PATCH 2865/4131] drm/amdkfd: Address logic error for actual_size

Return error if actual_size pointer is NULL in copy_mem_to_mem

Change-Id: Id9c4ff5824cce3017c1a8f34d1c6ea0c116d832b
Signed-off-by: Kent Russell <kent.russell@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c
index 1bff21b..869487b5 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c
@@ -2412,11 +2412,10 @@ int amdgpu_amdkfd_copy_mem_to_mem(struct kgd_dev *kgd, struct kgd_mem *src_mem,
 	struct dma_fence *fence = NULL;
 	int i, r;
 
-	if (!kgd || !src_mem || !dst_mem)
+	if (!kgd || !src_mem || !dst_mem || !actual_size)
 		return -EINVAL;
 
-	if (actual_size)
-		*actual_size = 0;
+	*actual_size = 0;
 
 	adev = get_amdgpu_device(kgd);
 	INIT_LIST_HEAD(&list);
-- 
2.7.4

